### PR TITLE
Update example YAML in encryption-at-rest topic

### PR DESCRIPTION
**Pull request by sftim**
Revise [Encrypting Secret Data at Rest](https://kubernetes.io/docs/tasks/administer-cluster/encrypt-data/) [[preview](https://deploy-preview-41574--kubernetes-io-main-staging.netlify.app/docs/tasks/administer-cluster/encrypt-data/)]
Split off from PR #33285

Tweak the manifests and other documents for more comments, easier reading, etc.

### CODE REVIEW

File: content/en/docs/tasks/administer-cluster/encrypt-data.md [hunk 1]
```diff
@@ -45,15 +45,24 @@ decrypt data stored in the etcd.
 ## Understanding the encryption at rest configuration
 
 ```yaml
+---
```
Comment (enj) [inline content/en/docs/tasks/administer-cluster/encrypt-data.md:48]: This looks a bit odd in the preview.

Comment (sftim) [reply content/en/docs/tasks/administer-cluster/encrypt-data.md:48]: I'll explain my thinking here:
- I'd like us (SIG Docs) to start all YAML that represents an _entire_ document with `---`
- YAML fragments should never start with `---`

The idea is that some readers can immediately tell which are which, and if not then we can one day have ways to automate providing useful hints.

Comment (reylejano) [reply content/en/docs/tasks/administer-cluster/encrypt-data.md:48]: starting a yaml with `---` is common practice and preferred by many

**Comment by reylejano** [comment]
For any one following up on this PR, I agree with the change to use port 443 for the annotation `kubeadm.kubernetes.io/kube-apiserver.advertise-address.endpoint` in the example because 443 is commonly used in production for the api server instead of 6443
See: https://kubernetes.io/docs/concepts/security/controlling-access/

> By default, the Kubernetes API server listens on port 6443 on the first non-localhost network interface, protected by TLS. In a typical production Kubernetes cluster, the API serves on port 443. 

/approve

**Pull request merged**